[doc][core][cgraph] Use with_tensor_transport API (#50273)
ray-project/ray: doc/source/ray-core/compiled-graph/overlap.rst

        with InputNode() as inp:
            branches = [sender.send.bind(shape, dtype, inp) for sender in senders]
            branches = [
                branch.with_tensor_transport(
                    transport="nccl", _static_shape=True, _direct_return=True
                )

>                 # Note that before ray version 2.42, use `with_type_hint()` instead.

Reviewer: ```suggestion
                # For any ray version before 2.42, use `with_type_hint()` instead.
```
